### DISCUSSION TITLE
How do I create a plugin to launch a console text application in a floating window?

### DISCUSSION THREAD

**Discussion by mjarmy** [neovim/neovim#29918] [Q&A] [answered]
I'm working on creating a terminal-based file manager application. I'd like to be able to launch it from inside nvim, with a command-line parameter specifying the nvim current directory so the application can start there.  When the application is done, I want it to return a path to a file (or directory) that nvim should then open in its current window.  I know I need to create a plugin to do this, but I am a plugin noob so I need some help.

Another way of putting what I want is: I'd like the experience of using my application to be exactly like it is for [fzf](https://github.com/junegunn/fzf) -- you launch the application using the current nvim directory, do some stuff in the floating window, and then when you are done nvim goes to a file that is returned from the application.

I took a look in the fzf [plugin directory](https://github.com/junegunn/fzf/blob/master/plugin/fzf.vim) since that seemed like an obvious place to figure out how to make this work, but that is a really complicated vim script, and so I don't think its helpful for a plugin beginner like me.  Surely there is an easier way to do this?  

I only care about getting this to work in neovim by the way -- I'm not interested in vim support.  I'd love to have a simple example to use of how to do this using a lua plugin.  Can anyone point me in the right direction?

For what its worth, I'm using the same [console library](https://github.com/gdamore/tcell) that fzf does, so I'm sure my application will be a proper terminal UI.

**Comment by zeertzjq** [comment, marked as answer]
For opening a terminal buffer in Nvim, see `:h termopen()`.

**Reply by mjarmy** [reply]
That's really all I need to get started, thanks
